source/geod24/concurrency.d

```diff
@@ -985,6 +986,161 @@ protected:
         bool notified;
     }
 
+    class Channel (T)
+    {
+        this(ulong size = 0)
+        {
+            this.size = size;
+            m_lock = new Mutex;
+        }
+
+        bool write(T val)
+        {
+            bool notify_peer, fiber_was_blocked = false;
+            ChannelQueueEntry qEnt;
+
+            synchronized (m_lock)
+            {
+                notify_peer = m_rxq.length > 0;
+
+                if (m_closed)
+                {
+                    return false;
+                }
+                else if (m_rxq.length > 0)
+                {
+                    qEnt = m_rxq[0];
+                    m_rxq = m_rxq.remove(0);
```

Comment:
```suggestion
                    m_rxq = m_rxq.popFront();
```

Can use range functions like this from `std.range`.

Reply:
But actually using an array as a queue is inefficient. It's going to cause a lot of reallocations, because the array keeps growing but its pointer also keeps being shifted to the right. It's better to use something like a `DList` from `std.container` instead.

Reply:
Yes, I am aware the storage is not optimal. I will check `DList` out. Thanks!

Reply:
It's ok to keep it as-is if you already plan to improve the storage later. I just wanted to make a suggestion for the future. :)